spring-data-relational/src/main/java/org/springframework/data/relational/core/sql/When.java

```diff
@@ -0,0 +1,53 @@
+package org.springframework.data.relational.core.sql;
+
+/**
+ * When segment for Case statement.
+ * <p>
+ * Results in a rendered condition: {@code WHEN <condition> THEN <value>}.
+ * </p>
+ *
+ * @author Sven Rienstra
+ * @since 3.4
+ */
+public class When extends AbstractSegment {
+    private final Expression condition;
```

Comment:
`condition` should be an actual `Condition`

Reply:
Note: there are other forms of `CASE` expressions, where you have an expression to start with and in each branch an expression to compare it to. See https://modern-sql.com/feature/case
But that is not what we are doing here.